index.js

```diff
@@ -49,7 +62,18 @@ getUsersCreditDetails(usersArray);
 // ***************************************************************************
 
 const genderView = users => {
-  // Your code goes here ...
+  const femaleUsers = [];
+  const maleUsers = [];
+  for (let user of users) {
+    const {firstName,lastName,gender} = user;
+    if(gender==='female') {
+      femaleUsers.push(`${firstName} ${lastName}`)
+    } else {
+      maleUsers.push(`${firstName} ${lastName}`)
+    }
```

Comment:
these two for loops are correct, but you could have made this in one map function, or, like the exercise wanted a filter .

```js
let femaleUsers = users.filter(item => item.gender === “female”)
let maleUsers = users.filter(item => item.gender === “male”)
```

and here you would have the arrays already split by gender. and then you could do the transformation you want by mapping over them or something else.

But its totally correct 💪